bindings/native/go/native.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+package stronghold
+
+import (
+	"errors"
+	"strings"
+	"unsafe"
+)
+
+// TODO: Clean up paths once we have a working build pipeline
+/*
+#cgo LDFLAGS: -Wl,-rpath,${SRCDIR}/../../../target/debug/ -L${SRCDIR}/../../../target/debug/ -L${SRCDIR}/dist/ -L${SRCDIR}/../dist/ -lstronghold_native
+#include "dist/stronghold_native.h"
+*/
+import "C"
+
+type StrongholdPointer *C.struct_StrongholdWrapper
+
+const SignatureSize = 64
+const PublicKeySize = 32
+
+func createSnapshot(snapshotPath string, key string) (StrongholdPointer, error) {
+	snapshotPathNative := C.CString(snapshotPath)
+	defer C.free(unsafe.Pointer(snapshotPathNative))
+
+	keyNative := C.CString(key)
+	defer C.free(unsafe.Pointer(keyNative))
+
+	ptr := unsafe.Pointer(C.stronghold_create(snapshotPathNative, keyNative))
+
+	if err := handlePtrError(ptr != nil); err != nil {
+		return nil, err
+	}
+
+	return StrongholdPointer(ptr), nil
+}
+
+func loadSnapshot(snapshotPath string, key string) (StrongholdPointer, error) {
+	snapshotPathNative := C.CString(snapshotPath)
+	defer C.free(unsafe.Pointer(snapshotPathNative))
+
+	keyNative := C.CString(key)
+	defer C.free(unsafe.Pointer(keyNative))
+
+	ptr := unsafe.Pointer(C.stronghold_load(snapshotPathNative, keyNative))
+
+	if err := handlePtrError(ptr != nil); err != nil {
+		return nil, err
+	}
+
+	return StrongholdPointer(ptr), nil
+}
+
+func destroyStronghold(strongholdPtr StrongholdPointer) {
+	C.stronghold_destroy_stronghold(strongholdPtr)
+}
+
+func destroyErrorPointer(ptr *C.char) {
+	C.stronghold_destroy_error(ptr)
+}
+
+func destroyDataPointer(ptr unsafe.Pointer) {
+	C.stronghold_destroy_data_pointer((*C.uchar)(ptr)) //nolint:typecheck
+}
+
+func generateED25519KeyPair(strongholdPtr StrongholdPointer, key string, recordPath string) ([PublicKeySize]byte, error) {
+	keyNative := C.CString(key)
+	defer C.free(unsafe.Pointer(keyNative))
```

Comment:
`C.free` will only release the memory held by the key, but not fill it with zeroes. Please clear the respective memory area beforehand.